pkg/registrymirror/registrymirrorsstore.go

```diff
@@ -0,0 +1,325 @@
+package registrymirror
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+	ciTypes "github.com/containers/image/v5/types"
+	"github.com/docker/distribution/reference"
+	configV1 "github.com/openshift/api/config/v1"
+	operatorV1Alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openshift/runtime-utils/pkg/registries"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/maputil"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// default path to store registries config.
+	defaultRegistriesPath = "/var/cache/stackrox/mirrors/registries.conf"
+
+	// default delay before writing the updated registries config.
+	defaultDelay = time.Millisecond * 300
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// Store defines an interface for interacting with registry mirrors.
```

Comment:
are these registry mirrors just an OpenShift concept? If not, should we make it clear somewhere that we are solely handling OpenShift mirrors?

Reply:
The custom resources currently being watched are specific to the OpenShift API and handled by the OpenShift `machine-config-operator`. 

However mirroring could potentially be configured outside of those custom resources and anywhere `CRI-O` is used (including *k8s) - so for right now, yes its OCP only - but in the future could be more than OCP.